### ISSUE TITLE
Unusual behaviour for response headers keys registry

### ISSUE THREAD

**Issue by fulldiver-ilya** [ProxymanApp/Proxyman#373]
### macOS Version: 10.15.2

Hey! Thanks a lot for an amazing tool, but I have a question...

I'm developing iOS application with Alamofire (most common 3-party network library) and using http-headers from our internal server responses. Alamofire provides case-insensitive keys for all headers (they are all lower-cased).
But there is a problem when I use Proxyman: after getting response that pass through Proxyman I have a case like `X-Custom-Header`. When I toggle-off proxy I get `x-custom-header` as I expect

Is any way to change or understand this behaviour?

**Comment by NghiaTranUIT** [comment]
Thank for the report @fulldiver-ilya. Technically, all Field Headers are case-insensitive (https://www.w3.org/Protocols/rfc2616/rfc2616-sec4.html#sec4.2), so the behavior of the app from end-users should not changed.

However, you're right that Proxyman implicitly capitalizes it. I will improve this behavior by:
- Keep the same key from the client or server.

I'm on it and send you hot-fix build in this week 👍

**Comment by NghiaTranUIT** [comment]
Hi there, so far from what I investigated that Proxyman doesn't support HTTP/2 since we're heavily relied on CocoaAsyncSocket, which isn't supported it.

I tested with a testing server:
+ If not enable SSL -> Proxyman is a ordinary Proxy -> The HTTPS connection is upgraded to HTTP/2 -> The header keys are lowercase from the server.
+ If enable SSL -> Proxyman does SSLHandshake, but refuse upgrading HTTP/2 from the server -> Use HTTP/1.1 -> The server returns uppercased header by default.

In the initial thought, It might be easy, but actually it's difficult since I have to find a way to support HTTP/2 in CocoaAsyncSocket.

---------------
About your case, I believe that your connection from iOS app is downgrade to HTTP/1.1 and the Header in the response from your server are uppercase => Proxyman sends back to the app => `X-Custom-Header`

Can we @fulldiver-ilya lowercase in your app again to make sure it works in both HTTP/1.1 and HTTP/2?

**Comment by fulldiver-ilya** [comment]
Sorry for late answer but yes, our server is working over http/2.0 with SSL and Proxyman downgrading answer to http/1.1

Proxyman answer:
<img width="111" alt="Screenshot 2020-01-15 at 17 18 39" src="https://user-images.githubusercontent.com/59820813/72441159-3e308280-37bb-11ea-9515-0d526b555633.png">

Charles answer:
<img width="237" alt="Screenshot 2020-01-15 at 17 16 46" src="https://user-images.githubusercontent.com/59820813/72441161-3e308280-37bb-11ea-8b31-5711b0384111.png">

Is it answer to your guess?

**Comment by NghiaTranUIT** [comment]
Yup, it's correct as I mention that Proxyman currently doesn't support HTTP/2. I'm looking on it to support HTTP/2 soon.

**Comment by NghiaTranUIT** [comment]
Hey @fulldiver-ilya , the good news that Proxyman 2.0.0 is re-written with Swift NIO (https://github.com/ProxymanApp/Proxyman/issues/500), which is capable of supporting HTTP/2. 
Let bump this feature forward 🙌

**Comment by fulldiver-ilya** [comment]
Hey @NghiaTranUIT its a great news! Congrats 🎉

**Comment by nunogoncalves** [comment]
Hello, 

We are seeing the same. an API is returning capitalized headers, and after passing by Proxyman they are lowercased. 
Is there any configuration to fix this?

**Comment by NghiaTranUIT** [comment]
Just wondering: Do your request/response use any debugging tools?

In the latest build, Proxyman is using HTTP/1.1, so all Headers are capitalized on the first character, and Proxyman doesn't modify any Header by default.

**Comment by nunogoncalves** [comment]
Ok, I think we got it. If we make the request using curl —http1.1 it shows us Auth-Token. If we set it for —http2.0 it lowercases  it. I'm unsure to what the correct behaviour should be. :D

**Comment by nunogoncalves** [comment]
So, our app, iOS is seeing auth-token (which seems to support the idea that it supports http 2.0). If we use Proxyman, it shows Auth-Token. If Proxyman supports http2, shouldn't it be auth-token instead? Maybe I'm missing something here.

**Comment by NghiaTranUIT** [comment]
In HTTP/1.1, it's Auth-Token (capitalized on the first character). In HTTP/2, all header is lowercase.
Proxyman hasn't implemented HTTP/2 yet, so all Header will be `Auth-Token`

**Comment by nunogoncalves** [comment]
Ah! Sorry! 

So I'm confused to this messagE: 

> Hey @fulldiver-ilya , the good news that Proxyman 2.0.0 is re-written with Swift NIO (#500), which is capable of supporting HTTP/2.

**Comment by NghiaTranUIT** [comment]
ah, yes, sorry for the misunderstanding. It means, Proxyman, powered by SwiftNIO, can support HTTP/2/ in the future. It's easier than the v1.0, which uses bunch of raw socket 😄 

HTT/2 is still in our road map 👍

**Comment by nunogoncalves** [comment]
Thanks for your answers. 
BTW, I got to say. This was by far the best open source experience I've ever had! It never crossed my mind that I'd get answers in 3 minutes! :D

**Comment by NghiaTranUIT** [comment]
Thanks. Glad to know that you really enjoy the app 😄 ❤️

**Comment by stodirascu** [comment]
+1 for the awesomeness of the app 🎉

And +1 to solve this issue. I've stumbled across it recently and baffled me a bit till I understood what's going on. Is there a way to upvote this to switch to HTTP/2 sooner than later?

**Comment by NghiaTranUIT** [comment]
Thanks folks to bump the HTTP/2. I will put it in the high priority queue to implement it 👍
